### PR TITLE
Fix edit icon behavior

### PR DESCRIPTION
**Pull request by misslivirose**
Fixes FB case 14082: "Create" icon is grayed out even when I do have rez rights in moving between domains.

Should also cover 14087

This also removed some old code that was relying on messages to determine if domains were changed instead of the Window.domainChanged signal. 

Also handles changes on the server side to editing permissions.

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/8885/app_12929_PR_2a055429cd9bdd1ce5bcbd45e255a9c2fd38d542.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/8885/HighFidelity-Beta-PR12929-2a055429cd9bdd1ce5bcbd45e255a9c2fd38d542.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/8885/HighFidelity-Beta-Interface-PR12929-2a055429cd9bdd1ce5bcbd45e255a9c2fd38d542.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/8885/HighFidelity-Beta-PR12929-2a055429cd9bdd1ce5bcbd45e255a9c2fd38d542.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/8885/HighFidelity-Beta-Interface-PR12929-2a055429cd9bdd1ce5bcbd45e255a9c2fd38d542.exe

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/8896/app_12929_PR_e795af03ddb573046d6849ef0f77625a8753d33a.apk

**Comment by misslivirose** [comment]
**Test Plan**
_Bug Check & Changing Domains_
1. Confirm that the repro steps in 14082 behave as expected when changing domains.

_Full Test_
1. Go to your Sandbox
2. Open up your domain settings. Add a specific permission for your user account with "Connect"  permissions and no rez rights. Save.
3. Confirm the icon is grayed out.
4. Check 'Rez' for your user account and Save. Observe that the icon is now enabled.
5. Open the Create menu. 
6.. Uncheck 'Rez' for your user account in the domain server settings and Save. 
7. Confirm that the menu closes and button disables. 
8. Repeat steps 4 - 7 for Rez Temp, Rez Certified, and Rez Certified Temp
9. Repeat test for tablet UI.

_Shapes Check_
1. Use Shapes app from the Marketplace and confirm that you are able to add items to the world when you have rez rights.

File: scripts/system/edit.js [hunk 1]
```diff
@@ -452,19 +452,45 @@ var toolBar = (function () {
...
     }
 
+    // Handles any edit mode updates required when domains have switched
+    function checkEditPermissionsAndUpdate() {
+        if ((createButton === null) || (createButton === undefined)) {
+            //--EARLY EXIT--( nothing to safely update )
+            return;
+        }
+
+        var hasRezPermissions = (Entities.canRez() || Entities.canRezTmp() || Entities.canRezCertified() || Entities.canRezTmpCertified());
```
Comment (thoys) [inline scripts/system/edit.js]: line length is over 128 characters, but it was before this PR so I won't fail it over that

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/8896/HighFidelity-Beta-PR12929-e795af03ddb573046d6849ef0f77625a8753d33a.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/8896/HighFidelity-Beta-Interface-PR12929-e795af03ddb573046d6849ef0f77625a8753d33a.dmg

**Comment by misslivirose** [comment]
build this please

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/8898/app_12929_PR_e795af03ddb573046d6849ef0f77625a8753d33a.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/8898/HighFidelity-Beta-PR12929-e795af03ddb573046d6849ef0f77625a8753d33a.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/8898/HighFidelity-Beta-Interface-PR12929-e795af03ddb573046d6849ef0f77625a8753d33a.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/8898/HighFidelity-Beta-PR12929-e795af03ddb573046d6849ef0f77625a8753d33a.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/8898/HighFidelity-Beta-Interface-PR12929-e795af03ddb573046d6849ef0f77625a8753d33a.exe

**Comment by qlimnavarro** [comment]
Pass @conklin94122 
Items in UI react according to permissions.

**Pull request merged**